packages/common-utils/src/queryParser.ts

```diff
@@ -666,50 +726,88 @@ async function nodeTerm(
       rangedTerm.term_min,
       rangedTerm.term_max,
       isNegatedField,
+      context,
     );
   }
 
   throw new Error(`Unexpected Node type. ${node}`);
 }
 
+function createSerializerContext(
+  currentContext: SerializerContext,
+  ast: lucene.BinaryAST | lucene.LeftOnlyAST,
+) {
+  // For syntax like `foo:(bar baz)` or `foo:("bar baz")`, the implicit field for the inner expression must be `foo`
+  if (ast.field && ast.parenthesized && ast.field !== IMPLICIT_FIELD) {
+    const fieldWithoutNegation = ast.field?.startsWith('-')
+      ? ast.field.slice(1)
+      : ast.field;
+
+    return {
+      ...currentContext,
+      implicitColumnExpression: fieldWithoutNegation,
```

Comment:
Ah, I see the logic here. Even though the parser doesn’t parse the field correctly, we still extract the field and assign it to implicitColumnExpression, which serves as the fallback behavior.
I wonder if the right fix would be to make the parser handle it correctly and produce the proper AST structure. I’m fine shipping these updates for now, but we’ll want to deprecate them once we migrate to the actual ClickHouse query parser